app/controllers/events_controller.rb

```diff
@@ -0,0 +1,73 @@
+class EventsController < ApplicationController
+  before_action :set_project
```

Comment:
just use  before_action :authenticate_user! in your controller, it will automatically redirect you to sign_in page, then you do not need to check if current_user is nil in your controller.